File: src/GroundZero.Application/Features/Appointments/DTOs/CreateAppointmentRequest.cs
namespace GroundZero.Application.Features.Appointments.DTOs;

public class CreateAppointmentRequest
{
    public int StaffId { get; set; }
    public DateTime ScheduledAt { get; set; }
    public int DurationMinutes { get; set; }
    public string? Notes { get; set; }
}

File: groundzero_mobile/lib/features/appointments/models/create_appointment_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'create_appointment_request.freezed.dart';
part 'create_appointment_request.g.dart';

@freezed
class CreateAppointmentRequest with _$CreateAppointmentRequest {
  const factory CreateAppointmentRequest({
    required int staffId,
    required DateTime scheduledAt,
    required int durationMinutes,
    String? notes,
  }) = _CreateAppointmentRequest;

  factory CreateAppointmentRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateAppointmentRequestFromJson(json);
}
